### ISSUE TITLE
Error while deploying more than one model in batch mode

### ISSUE THREAD

**Issue by mohsinkhalid122** [ThinkBigAnalytics/AoaDockerComposeDemos#29]
Hi,
 I'm facing an I/O error while deploying more than one model in batch mode. You can deploy one model if you try to deploy another model this 'file exists' error appears. This error also occurs in the previous versions as well. Below is the error.

mkdir -p /app/aoa/aoa-serving-engines/ci/local/../../../dags
mkdir: cannot create directory â€˜/app/aoa/aoa-serving-engines/ci/local/../../../dagsâ€™: File exists
make: *** [Makefile:92: scheduler_dag_template] Error 1
report_error 2
'[' 2 == 0 ']'
ERR_MESSAGE='error in aoa-serving-engines/ci/local/local-deployer.sh'
curl -X POST -u admin:admin 'http://localhost:8080/api/message/jobevents?type=topic' -H 'Content-Type: application/json' -d '{
                "jobId": "56040b2f-8397-46e7-80b0-b6c3229e4f06",
                "projectId": "47a89129-e3a3-4e4c-a307-338824508afd",
                "ownerId": "admin",
                "status": "ERROR",
                "metadata": {
                    "source": "local deployer",
                    "message": "error in aoa-serving-engines/ci/local/local-deployer.sh"
                }
            }'
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   405    0     0  100   405      0  27000 --:--:-- --:--:-- --:--:-- 28928
exit 2

**Comment by willfleury** [comment]
Something very strange in this error is the special characters in it - see the end of the file path. This should not exist. Also you can see that we do a mkdir -p which means that it will only try to create it if it does not already exist which means you should never get a file exists error unless its a file vs a dir. 

![image](https://user-images.githubusercontent.com/10163928/102981293-096b9600-4509-11eb-9255-091f26de9923.png)

Actually, @mohsinkhalid122 I have a feeling i know what is wrong. Can you do the following locally

`rm -rf /tmp/scheduler`

Basically, that will cleanup the folder that is used to hold the dags between the containers. We changed the structure of this in different versions and I think that might be causing an issue for you. Also, because this is a tmp folder, if you have not updated it to use a non tmp and you are running this process long term, the files may be removed (as its tmp). That isn't a bug but rather a misconfiguration on your deployment (a real deployment shouldn't use tmp). We can help you with that in the new year but for now try the "rm" to cleanup

**Comment by Christian-TD** [comment]
Hi @mohsinkhalid122, adding to what @willfleury pointed out above, you can also try to run this command in order to cleanup the whole demo environment:

`docker-compose down
docker rmi $(docker images -a --format '{{.Repository}}:{{.Tag}}' | grep 'thinkbiganalytics/aoa-')
rm -rf /tmp/scheduler
rm -rf /tmp/aoa-logs
docker-compose up`

Remember that the demo environment can only run one restful container at a time, so when deploying a second restful model, the previous one will be deleted. In batch mode, the auto removal of the containers is now fixed, so now everything should be working fine in local batch mode.

**Comment by willfleury** [comment]
@mohsinkhalid122 can you confirm cleaning up the filesystem fixed the issue? Or do you still have the issue?

**Comment by mohsinkhalid122** [comment]
@willfleury so I have one model deployed and I ran this command
`rm -rf /tmp/scheduler`
This didn't work and I got the same error again.
@Christian-TD Your solution would work if I restart AOps.  I want to deploy another model without restarting AOps.
